sklearn/ensemble/forest.py

```diff
@@ -104,7 +130,9 @@ def _parallel_build_trees(tree, forest, X, y, sample_weight, tree_idx, n_trees,
         else:
             curr_sample_weight = sample_weight.copy()
 
-        indices = _generate_sample_indices(tree.random_state, n_samples)
+        n_bootstrap_samples = _get_n_bootstrap_samples(n_samples, max_samples)
```

Comment:
But surely this can happen as soon as you have the data, along with other validation rather than when building

Reply:
Yep, no sense in computing it `n_estimators` times if it doesn't change! Done in: https://github.com/scikit-learn/scikit-learn/pull/14682/commits/4fa10a9f6d2a41d014ce7b8451b814429070f933